core/src/test/scala/org/apache/spark/storage/BlockManagerReplicationSuite.scala

```diff
@@ -494,7 +494,9 @@ class BlockManagerProactiveReplicationSuite extends BlockManagerReplicationBehav
 
     val newLocations = master.getLocations(blockId).toSet
     logInfo(s"New locations : $newLocations")
-    assert(newLocations.size === replicationFactor)
+    eventually(timeout(5 seconds), interval(10 millis)) {
+      assert(newLocations.size === replicationFactor)
```

Comment:
line 495 needs to be in here too -- otherwise you're continually checking the same set of locations